sphinx/domains/python.py

```diff
@@ -495,7 +495,15 @@ def handle_signature(self, sig: str, signode: desc_signature) -> Tuple[str, str]
 
         sig_prefix = self.get_signature_prefix(sig)
         if sig_prefix:
-            signode += addnodes.desc_annotation(str(sig_prefix), '', *sig_prefix)
+            if type(sig_prefix) is str:
+                logger.warning(
```

Comment:
Could you use `warnings.warn()` instead? It can describe what version this fallback will be dropped.

Reply:
Sure. Removed in v5 or v6?
Is there a way to automatically attach the location that triggered this? (the ``location=signode`` that ``logger.warning()`` accepts)?